pandas/_libs/tslibs/conversion.pyx

```diff
@@ -646,7 +646,7 @@ cdef _TSObject convert_str_to_tsobject(str ts, tzinfo tz,
         reso = get_supported_reso(out_bestunit)
         return convert_datetime_to_tsobject(dt, tz, nanos=nanos, reso=reso)
 
-    return convert_datetime_to_tsobject(dt, tz)
+    return convert_datetime_to_tsobject(dt, tz, nanos=0, reso=NPY_FR_us)
```

Comment:
This general case probably cannot be hardcoded

Reply:
Looks like the only cases left to handle here are 'now' and 'today', all other branches have returned.

I think we could combine these two cases and then return, to make the code more clear.
https://github.com/pandas-dev/pandas/blob/d6fdbee9f31f621f5177f65abc4525c822ab2e97/pandas/_libs/tslibs/conversion.pyx#L597-L605